tests/backup/backup_psa_test.go

```diff
@@ -548,3 +548,440 @@ var _ = Describe("{EnableNsAndClusterLevelPSAWithBackupAndRestore}", Label(TestC
+
+	JustBeforeEach(func() {
+		// TODO: Need to update testcase ID
+		StartPxBackupTorpedoTest("RestoreFromHigherPrivilegedNamespaceToLower", "Restore from higher Privileged to lower Privileged namespace", nil, 299239, Sn, Q2FY25)
+
+		log.InfoD("Deploy applications")
+		scheduledAppContexts = make([]*scheduler.Context, 0)
+		// Define maps to store unique namespaces
+		baselineNamespaceMap := make(map[string]struct{})
+		restrictedNamespaceMap := make(map[string]struct{})
+		psaApp := make([]string, 0)
+
+		//Deploy multiple applications on multiple namespace on restricted and baseline namespaces
+		for _, psaLevel := range []string{"restricted", "baseline"} {
+
+			if strings.Contains(psaLevel, "restricted") {
+				log.InfoD("The app list at the start of the testcase is %v", originalAppList)
+				for _, app := range originalAppList {
+					psaApp = append(psaApp, PSAAppMap[app])
+				}
+				Inst().AppList = psaApp
+				log.Infof("The PSA app list for restricted namespace is %v", psaApp)
+			}
+			label["pod-security.kubernetes.io/enforce"] = psaLevel
+			err := Inst().S.RescanSpecs(Inst().SpecDir, Inst().V.String())
+			log.FailOnError(err, "Failed to rescan specs from %s for storage provider %s", Inst().SpecDir, Inst().V.String())
+
+			for i := 0; i < len(originalAppList); i++ {
+				taskName := fmt.Sprintf("%s-%d-%s", psaLevel, i, RandomString(10))
+				namespace := fmt.Sprintf("%s-%d-%s", psaLevel, i, RandomString(10))
+				appPrivilegeToNsMap[psaLevel] = namespace
+				err = CreateNamespaceAndAssignLabels(namespace, label)
+				dash.VerifyFatal(err, nil, "Creating namespace and assigning labels")
+				appContexts := ScheduleApplicationsOnNamespace(namespace, taskName)
+				for _, ctx := range appContexts {
+					ctx.ReadinessTimeout = AppReadinessTimeout
+					namespace := GetAppNamespace(ctx, taskName)
+					scheduledAppContexts = append(scheduledAppContexts, ctx)
+					AppContextsMapping[namespace] = ctx
+					if strings.Contains(namespace, "restricted") {
+						restrictedScheduledAppContexts = append(restrictedScheduledAppContexts, ctx)
+						// Add to restrictedNamespaceMap if not already present
+						if _, ok := restrictedNamespaceMap[namespace]; !ok {
+							restrictedNamespaceMap[namespace] = struct{}{}
+						}
+					} else if strings.Contains(namespace, "baseline") {
+						baselineScheduledAppContexts = append(baselineScheduledAppContexts, ctx)
+						// Add to baselineNamespaceMap if not already present
+						if _, ok := baselineNamespaceMap[namespace]; !ok {
+							baselineNamespaceMap[namespace] = struct{}{}
+						}
+					}
+				}
+			}
+			// Convert map keys to slice to store baseline namespaces
+			baselineNamespaceList = make([]string, 0, len(baselineNamespaceMap))
+			for ns := range baselineNamespaceMap {
+				baselineNamespaceList = append(baselineNamespaceList, ns)
+			}
+
+			// Convert map keys to slice to store restricted namespaces
+			restrictedNamespaceList = make([]string, 0, len(restrictedNamespaceMap))
```

Comment:
Can we make a local function for this and use it here?

Reply:
done